### ISSUE TITLE
Cannot create an account

### ISSUE THREAD

**Issue by Okamoi** [turtl/tracker#283]
Hi,

Using Turtl 0.7.2.5 on Windows 7 64-bit, I can't create an account. Since I don't have one already, I can't test if the issue is specific to account creation or not. My firewall allows all connections and so does my router. Firefox can reach the server URL displayed in Turtl's UI, `https://apiv3.turtlapp.com/`, and gets a `{"greeting":"Hi."}` JSON response as expected.

When I click on "Create an account" with a valid bogus email, Turtl connects to `members.linode.com` successfully over HTTPS, then `apps.identrust.com/roots/dstrootcax3.p7c` and downloads a 803 bytes file over HTTP.

Then no other connection is attempted, and this error pops up within Turtl's UI:

```
Il y a eu un problème lors de l’engistrement de ce compte: 
https://apiv3.turtlapp.com/users: an error occurred trying to connect: 
Une chaîne de certificats a été traitée mais s’est terminée par un certificat racine qui n’est pas approuvé par le fournisseur d’approbation. 
(os error -2146762487)
```

Translates to: Root certificate not approved by...approval provider, I suppose.

Curated logs, from app start to account creation error:

```
[INFO][turtl_core] main::init() -- init with user config {"data_folder":"C:\\Users\\*****\\AppData\\Roaming\\Turtl\\core","api":{"client_version_string":"desktop/0.7.2.5","proxy":null},"openssl_cert_file":"D:\\*****\\Turtl\\resources\\app\\scripts\\resources\\cacert.pem"}
[INFO][turtl_core] main::init() -- created data folder: C:\Users\*****\AppData\Roaming\Turtl\core
[INFO][turtl_core] main::start() -- begin
[INFO][turtl_core] main::start() -- locking data dir: C:\Users\*****\AppData\Roaming\Turtl\core/run.lock
[DEBUG][turtl_core::models::model] model -- set_client_id(): *******************
[INFO][turtl_core::messaging] messaging::start() -- main loop
[INFO][turtl_core::messaging] messaging::ui_event() -- messaging:ready
[INFO][turtl_core::dispatch] dispatch(1): app:api:set-endpoint
[INFO][turtl_core::dispatch] dispatch(2): app:api:get-endpoint
[INFO][turtl_core::dispatch] dispatch(3): app:api:get-old-endpoint
[INFO][turtl_core::dispatch] dispatch(4): app:get-log
[INFO][turtl_core::dispatch] dispatch(6): app:api:get-old-endpoint
[INFO][turtl_core::dispatch] dispatch(5): app:api:get-endpoint
[INFO][turtl_core::dispatch] dispatch(8): app:api:get-old-endpoint
[INFO][turtl_core::dispatch] dispatch(7): app:api:get-endpoint
[INFO][turtl_core::dispatch] dispatch(9): app:api:set-endpoint
[INFO][turtl_core::dispatch] dispatch(10): user:join
[INFO][turtl_core::models::user] user::generate_auth() -- generating v0 auth
[DEBUG][turtl_core::api] api::req() -- begin: POST /users
[DEBUG][turtl_core::api] api::call() -- creating new client with cachekey 
[DEBUG][turtl_core::api] api::call() -- req: POST https://apiv3.turtlapp.com/users
[DEBUG][turtl_core::api] api::call() -- call error: {"err":{"message":"https://apiv3.turtlapp.com/users: an error occurred trying to connect: Une chaîne de certificats a été traitée mais s’est terminée par un certificat racine qui n’est pas approuvé par le fournisseur d’approbation. (os error -2146762487)","type":"generic"},"file":"src\\api.rs","line":138,"wrapped":true}
[INFO][turtl_core::dispatch] dispatch(11): app:api:get-endpoint
[INFO][turtl_core::dispatch] dispatch(12): app:api:get-old-endpoint
[INFO][turtl_core::dispatch] dispatch(13): app:get-log
```

**Comment by rubbingalcoholic** [comment]
Sorry you ran into this. Can you verify that your system clock is set
correctly?

**Comment by Okamoi** [comment]
Yes, logs give time stamps like `2019-02-04T15:30:49`, just like the Windows clock in the tray bar.

I have to add that my Windows 7 install is a couple years out of date... If you think that's the culprit, I'll come back in some days or weeks, as I intend to reinstall Windows from scratch instead of merely updating. I figured I might as well do things thoroughly. But despite the lack of update, my system is very clean, works fine, not corrupted or anything.

> Sorry you ran into this.

It's okay, I'm willing to spend some time on it :)    
What got me was the tag filtering and the way notes are presented, plus the fact that it can all be local (although setting up a server ain't too straightforward and there's no purely offline mode yet).

**Comment by rubbingalcoholic** [comment]
Another thing worth checking is I believe we're using Let's Encrypt for the
SSL generation. If I recall correctly that might not be natively supported
in Windows 7, but there may be a workaround.

**Comment by Okamoi** [comment]
I'm not competent in certificate matters, but if you need me to check or try something specific do tell. Though I think you're talking to your future self so he is reminded to follow that lead :)

**Comment by orthecreedence** [comment]
I think it's very likely that Let's Encrypt is not "trusted" by the root certificate list in Windows 7. 

@Okamoi can you try going to https://apiv3.turtlapp.com in Internet Explorer?

My guess is that Firefox has a set of certificates that ship with it outside of the operating system it uses (and among them is Let's Encrypt), and Internet Explorer does not ship with those certificates so it should break. If IE does break, it would point to an issue in the operating system, and it might be helped by running updates.

**Comment by Okamoi** [comment]
Hah! First time I'll be running IE on this computer. Good find ! I indeed get a certificate error. When I click on the button that ignores the error, the connection gets through.

If you know off the top of your mind a workaround as mentioned earlier, I'll be glad to hear it, otherwise I'll be back once my computer is refreshed to confirm that it works.

It's a lucky find because I bumped into quite a similar issue trying to run another program a month ago, and the last hope I had was to update my computer. I was thinking it relied on some updated version of some DLL or whatever, but the culprit is very likely to be a certificate issue. So double thanks.

Make that triple. Thanks for your time :)

**Comment by orthecreedence** [comment]
Thanks for trying IE! So I did a little digging, and found this:

https://www.poweradmin.com/help/sslhints/ie.aspx

I think you could install the certificate from here: https://letsencrypt.org/certs/isrgrootx1.pem.txt (found on this page: https://letsencrypt.org/certificates/).

If that doesn't work, my next best recommendation would be to run the latest updates since I'm fairly certain Windows manages its root certificates via Windows Update. I'm going to close this issue for now, but feel free to post back and let us know how it goes =].

**Comment by Okamoi** [comment]
Hi,

So I tried installing this certificate as a trusted root by making a .cer file that's merely the text provided in your link, and it didn't help.

Then I reinstalled Windows 7 64-bit SP1 from scratch with the Convenience Rollup update and ran Windows update on top, with February 2019 updates and whatever else was proposed by Microsoft, along with up to date .NET Framework 4.7.2, an updated IE 11, and various Visual C++ Redistributables (up to 2017, the latest).

The error is still there : 

```
2019-03-05T16:38:08 - [INFO][turtl_core::dispatch] dispatch(17): app:api:set-endpoint
2019-03-05T16:38:08 - [INFO][turtl_core::dispatch] dispatch(18): user:join
2019-03-05T16:38:08 - [INFO][turtl_core::models::user] user::generate_auth() -- generating v0 auth
2019-03-05T16:38:08 - [DEBUG][turtl_core::api] api::req() -- begin: POST /users
2019-03-05T16:38:08 - [DEBUG][turtl_core::api] api::call() -- req: POST https://apiv3.turtlapp.com/users
2019-03-05T16:38:09 - [DEBUG][turtl_core::api] api::call() -- call error: {"err":{"message":"https://apiv3.turtlapp.com/users: an error occurred trying to connect: Une chaîne de certificats a été traitée mais s’est terminée par un certificat racine qui n’est pas approuvé par le fournisseur d’approbation. (os error -2146762487)","type":"generic"},"file":"src\\api.rs","line":138,"wrapped":true}
2019-03-05T16:38:37 - [INFO][turtl_core::dispatch] dispatch(19): app:api:get-endpoint
2019-03-05T16:38:37 - [INFO][turtl_core::dispatch] dispatch(20): app:api:get-old-endpoint
2019-03-05T16:38:46 - [INFO][turtl_core::dispatch] dispatch(21): app:get-log
```

My computer is [...] but works perfectly. It's a laptop with [specs edited out]. Windows and all the software installed is fully up to date.

**Comment by Okamoi** [comment]
Note: It is apparently the exact same issue, Internet Explorer can't connect to `https://apiv3.turtlapp.com/`.

Even installing the Let's Encrypt certificate you linked to on my now up to date computer does not work.

**Comment by Okamoi** [comment]
Just FYI, problem solved!

It appears that the culprit was a Group Policy tweak, coupled with the wrong certificate installed. It was not ISRG Root X1 that I needed, but the intermediate certificate `Let’s Encrypt Authority X3 (Signed by ISRG Root X1)`, which for some reason works even though Root X1 isn't installed, IIRC.

Now this would never have been a problem without the Group Policy at `gpedit.msc > Computer configuration > System > Internet Communication Management > Internet Communication settings > Turn Off Automatic Root Certificates Update` being set to Enabled.

I set it like that years ago, before TLS was widespread, because I considered that Windows update should deal with certificates on manually requesting a general OS update. I did not want Microsoft to be contacted and learn stuff about where I connect to.

Turns out Windows update does not really update certificates and they are instead being installed on the fly when an unknown one is encountered and it is seen as valid by Microsoft's servers, unless this group policy is enabled.

So I kept it enabled but now I know how to diagnose the issue it can create. It appears that the other program I talked about was also in need of a certificate, even though its error message was much less helpful than Turtl's and it was not a Let's Encrypt one.


Thanks for your patience, I would have had a hard time understanding the issue without you. I'd have thought it was time to buy a new laptop, which I absolutely don't need since this one can even play some modern games *on high*.
